content/blog/multi-cluster-consistency-models.md

```diff
@@ -0,0 +1,381 @@
+```
+It may take up to a couple of seconds for the recovery to complete then check that the message is now there in 'central'
+```bash
+nats --context sc-central stream view foo
+```
+
+### Killing two regions to go down at once and simulating a split brain
+The two failure scenarios are similar and related: a split brain from the point of view of the region getting isolated is no different from both of the other two regions going down at the same time.
+
+The difference being that in the case of split brain, the two other regions that can still see each other continue to operate normally (including processing new 'writes') and the isolated regions ends up in the same 'limited' mode of operation as in the case when two regions do down at the same time.  
+
+As soon as the network partition gets resolved or as the missing regions come back up the two parts of the brain will replicate missed messages between themselves and eventually become consistent again (though not necessarily in the same order).
+
+In the case of two regions going down at the same time or of being the smaller part of the split brain the remaining region can still operate but in a 'limited' fashion, as not all functionality will be available since there will be an inability for the remaining nodes to elect a JetStream 'meta leader'.
+- Publications to the stream _will still work_, the only way publications to stream in a regions would stop working is if 2 of the 3 servers in the region (or 2 out of 5) go down at the same time.
+- Get operations (e.g. what the KV 'get' operation uses) _will still work_.
+- Getting messages from already existing consumers (at the time the second regions goes down) on the stream _will still work_, and locally published messages will be seen in the 'read' stream right away.
+- However, creating new consumers (or new Streams) _will not work_.
+
+First kill both 'west' and 'east'
+```bash
+source killwest; source killeast
+```
+Publish a new message on 'central' (as if it was isolated region)
+```bash
+nats --context sc-central req foo.test 'Hello world from the central region'
+```
+Then bring down the 'central' region and 'east and 'west' back up
+```bash
+source killcentral; source startwest; source starteast
+```
+Wait up to a couple seconds and publish another message from one of those two regions
+```bash
+nats --context sc-east req foo.test 'Hello again from the east region'
+```
+Check you can create a new consumer and see that message from the other region
+```bash
+nats --context sc-west stream view foo
+```
+And finally resolve the split brain by restarting 'central'
+```bash
+source startcentral
+```
+After a few seconds you can see that all the messages where are now present in all the 'read' streams, though not necessarily in the same order by comparing the output of
+```bash
+nats --context sc-west stream view foo
+```
+With
+```bash
+nats --context sc-central stream view foo
+```
+
+# What if you still want global ordering?
```

Comment:
There is some controversy with using the word 'master' - we may want to switch to 'main' or 'primary'.

Reply:
@jnmoyne Could you add the 'About the Author' section at the end, please?

Reply:
yeah maybe reword this section to highlight that you have read streams mirrors that are following the original stream

Reply:
Changed